client/qiskit_serverless/core/clients/LocalClient.py

```diff
@@ -0,0 +1,230 @@
+                )
+        arguments = arguments or {}
+        env_vars = {
+            **(saved_program["env_vars"] or {}),
+            **{OT_PROGRAM_NAME: saved_program["title"]},
+            **{"PATH": os.environ["PATH"]},
+            **{ENV_JOB_ARGUMENTS: json.dumps(arguments, cls=QiskitObjectsEncoder)},
+        }
+
+        with Popen(
+            ["python", saved_program["working_dir"] + saved_program["entrypoint"]],
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            universal_newlines=True,
+            env=env_vars,
+        ) as pipe:
+            status = "SUCCEEDED"
+            if pipe.wait():
+                status = "FAILED"
+            output, _ = pipe.communicate()
+        results = re.search("\nSaved Result:(.+?):End Saved Result\n", output)
+        result = ""
+        if results:
+            result = results.group(1)
+
+        job = Job(job_id=str(uuid4()), client=self)
+        entry = {"status": status, "logs": output, "result": result, "job": job}
+        self._jobs[job.job_id] = entry
+        return job
+
+    def status(self, job_id: str):
+        return self._jobs[job_id]["status"]
+
+    def stop(self, job_id: str, service: Optional[QiskitRuntimeService] = None):
+        """Stops job/program."""
+        return f"job:{job_id} has already stopped"
+
+    def result(self, job_id: str):
+        return self._jobs[job_id]["result"]
+
+    def logs(self, job_id: str):
+        return self._jobs[job_id]["logs"]
+
+    def filtered_logs(self, job_id: str, **kwargs):
+        """Return filtered logs."""
+        raise NotImplementedError
+
+    #########################
+    ####### Functions #######
+    #########################
+
+    def upload(self, program: QiskitFunction):
+        # check if entrypoint exists
+        if not os.path.exists(os.path.join(program.working_dir, program.entrypoint)):
+            raise QiskitServerlessException(
+                f"Entrypoint file [{program.entrypoint}] does not exist "
+                f"in [{program.working_dir}] working directory."
+            )
+        self._patterns.append(
+            {
+                "title": program.title,
+                "provider": program.provider,
+                "entrypoint": program.entrypoint,
+                "working_dir": program.working_dir,
+                "env_vars": program.env_vars,
+                "arguments": json.dumps({}),
+                "dependencies": json.dumps(program.dependencies or []),
+            }
+        )
+        return program.title
+
+    def get_programs(self, **kwargs) -> List[QiskitFunction]:
+        """Returns list of programs."""
+        return [
+            QiskitFunction(
+                program.get("title"),
+                provider=program.get("provider", None),
+                raw_data=program,
+                client=self,
+            )
+            for program in self._patterns
+        ]
+
+    def get_program(
```

Comment:
I think this is a new API. My comment will be similar to my previous one. I'm fine with `get_program` but we need to maintain compatibility by now and support the old `get` method too, @korgan00 .

Reply:
Similar to my feedback about `get_job_by_id `, I don't think this problem is solved either. We can take a look tomorrow together if there is any doubt 👍